### PR TITLE
refactor(iot-dev): Add proxy support for file upload

### LINKED ISSUE #1037: Unable to upload file with HTTP proxy
- **OS, version, SKU and CPU architecture used:** <VERSION> Windows 10 Desktop x64
- **Application's .NET Target Framework :** <NETVERSION> netcoreapp2.2
- **Device:** <DEVICETYPE> Laptop (Simulated Device)
- **SDK version used:** <VERSION> Microsoft.Azure.Devices.Client 1.21.0

## Description of the issue:
It is not possible to upload file with proxy by use of `DeviceClient.UploadToBlobAsync`.
I found a similar question in Stack Overflow, but it has not beed resolved.
[Azure IoT Hub, File Upload through proxy?](https://stackoverflow.com/questions/54776386/azure-iot-hub-file-upload-through-proxy)

## Code sample exhibiting the issue:
```cs
var proxy = new WebProxy("http://proxy.example.com:3128");
var deviceClient = DeviceClient.CreateFromConnectionString("ConnectionStringForIoTHub", new ITransportSettings[]
{
    new AmqpTransportSettings(TransportType.Amqp_WebSocket_Only)
    {
        Proxy = proxy
    }
});

// DeviceTwin update works correctly
await deviceClient.GetTwinAsync().ConfigureAwait(false);
await deviceClient.UpdateReportedPropertiesAsync(new TwinCollection { ["status"] = "connected" }).ConfigureAwait(false); 

// File upload does not work correctly
var filePath = "test.log";
var blobName = "test.log";
using (var stream = new FileStream(filePath, FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
{
    await deviceClient.UploadToBlobAsync(blobName, stream).ConfigureAwait(false); 
}
```

### PR DESCRIPTION
**Pull request by timtay-microsoft**
No new APIs, just using the proxy settings configured in the client already if any are present.

Fixes #1037

### CODE REVIEW

File: iothub/device/src/InternalClient.cs [hunk 1]
```diff
@@ -757,7 +761,13 @@ public Task UploadToBlobAsync(string blobName, Stream source, CancellationToken
                 var context = new PipelineContext();
                 context.Set(_productInfo);
 
-                var transportSettings = new Http1TransportSettings();
+                //Use the proxy of the first transport settings if there is a proxy, regardless of protocol
```
Comment (drwill-ms) [inline iothub/device/src/InternalClient.cs:764]: space after //

Comment (timtay-microsoft) [reply iothub/device/src/InternalClient.cs:764]: Is there a way to make codemaid do that for us?

Comment (drwill-ms) [reply iothub/device/src/InternalClient.cs]: Might be. Take a look at the options dialog.

**Pull request merged**